packages/kbn-unified-data-table/src/components/row_height_settings.tsx

```diff
@@ -34,24 +38,21 @@ export function RowHeightSettings({
   label,
   rowHeight,
   rowHeightLines,
-  compressed,
   onChangeRowHeight,
   onChangeRowHeightLines,
   maxRowHeight,
   ['data-test-subj']: dataTestSubj,
 }: RowHeightSettingsProps) {
+  const prevRowHeightRef = useRef<RowHeightMode | undefined>();
+  // if rowHeightLines is -1 (auto mode) we can't populate input with -1
+  const initialRowHeightLines = rowHeightLines && rowHeightLines > 0 ? rowHeightLines : 3;
```

Comment:
This is to avoid situation like this:
1) You switch mode from `custom` to `auto` (so real lines count is -1, but input keeps previous custom value - as expected)
1.1) You can freely change from custom to auto and vice versa, without loosing input value tied to custom mode
2) While having `auto` mode on (-1 value stored in the store) you refresh the page
3) `rowHeightLines` coming via props is -1, so input is populated with -1 and switching between `auto` and `custom` is broken (input is disabled when in `auto` mode)

I see the room for improvement there though (which is keeping the input value in store as well), because I see one minor inconvenience in scenario like this:
1) You start from `custom` mode with default value `3`.
2) You change your custom value to eg. `7`
3) If you refresh page now, `7` is stored as `rowHeightLines` and will be correctly restored as a numeric input value
4) However if you switch to `auto`, then refresh page - input value falls back to default `3` (in order to not be assigned to -1), instead of `7`

Reply:
Also it can be overengineering, as it's my first a slightly deeper dive 😁